option cc_generic_services = true;
option java_generic_services = true;
option py_generic_services = true;

message GetRequest {
    required int32 key = 1;
}

message GetResponse {
	required string value = 1;
}

message PutRequest {
    required int32 key = 1;
    required string value = 2;
}

message PutResponse {
	required bool result = 1;
}

message GRRequest {
	required int32 start = 1;
	required int32 end = 2;
}

message GRResponse {
	required bool result = 1;
	repeated string value = 2;
}

message TxRequest {
	required string args = 1;
}

message TxResponse {
	message map {
		required int32 key = 1;
		required string value = 2;
	}
	repeated map retValue = 1;
}

service DbService
{
	rpc get(GetRequest) returns (GetResponse);
	rpc put(PutRequest) returns (PutResponse);
	rpc getrange(GRRequest) returns (GRResponse);
	rpc execTx(TxRequest) returns (TxResponse);
}